docs/How-to-add-new-gem-to-omsagent-bundle.md

```diff
@@ -0,0 +1,49 @@
+## How to add a new gem to omsagent install bundle
+
+In order to add a new Ruby gem into OMSAgent bundle when you have the new_gem_name.gem file on you local system you need to follow these steps:
+1. Update the private branch:
+* Clone Microsoft/Build-OMS-Agent-for-Linux github repo
+* Copy the new_gem_name.gem to Build-OMS-Agent-for-Linux//omsagent/source\ext\gems folder.
+* Navigate to Build-OMS-Agent-for-Linux//omsagent/build and put modified commands in buildRuby.sh
+(Example ${RUBY_DESTDIR}/bin/gem install ${BASE_DIR}/source/ext/gems/new_gem_name.gem)
+* Run buildRuby.sh 100
+* It will build the new changes for ruby under intermediate/*/100/Ruby.
+* Run ../installer/scripts/genRubyInstaller.sh.
+* It will create a new file containing changes for ruby.data in /tmp/installBuilder-ruby-data-$USER
+* Diff /tmp/installBuilder-ruby-data-$USER with the rub.data file from Build-OMS-Agent-for-Linux/omsagent/installer/datafiles/.
+* Observe only the new gem related records are added. Replace installer/datafiles/ruby.data with the content of installBuilder-ruby-data-$USER file
+
+2. Building the new changes privately:
+* Run ./configure the build environment with --enable-ulinux
+* make
+Install the private bundle and validate that /opt/microsoft/omsagent/ruby has the new gem changes.
+
+In order to add a new Ruby gem into OMSAgent bundle when you do not have the new_gem_name.gem file on you local system you need to follow these steps:
+1. Get the new_gem_name.gem file
+* Install OMSAgent bundle on you local machine.
+* Navigate to /opt/microsoft/omsagent/ruby and run gem install new_gem_name
+* Clone Microsoft/Build-OMS-Agent-for-Linux github repo
+* Find the newly installed new_gem_name.gem file from /opt/microsoft/omsagent/ruby path and copy it to Build-OMS-Agent-for-Linux//omsagent/source/ext/gems.
+* Navigate to Build-OMS-Agent-for-Linux/omsagent/installer/scripts and update genRubyInstaller.sh with SOURCE_DIR to be /opt/microsoft/omsagent/ruby
+* Run genRubyInstaller.sh. 
+* It will create a new file containing changes for ruby.data in /tmp/installBuilder-ruby-data-$USER
+* Diff /tmp/installBuilder-ruby-data-$USER with the rub.data file from Build-OMS-Agent-for-Linux/omsagent/installer/datafiles/.
+* Observe only the new gem related records are added. Replace installer/datafiles/ruby.data with the content of installBuilder-ruby-data-$USER file
+* Uninstall the agent (required to be able to build the agent from your private github depot)
+* Put  modified commands to Build-OMS-Agent-for-Linux/omsagent/build/buildRuby.sh
+(Example ${RUBY_DESTDIR}/bin/gem install ${BASE_DIR}/source/ext/gems/new_gem_name.gem)
```

Comment:
This is very long, lots of potential places to get confused. You have a real life example, I suggest you use it and give a nice, readable page, explaining (in real life) how you might add the MySQL gem into OMS.

Real life examples that they can read and follow, with real life screen output, is much easier to understand and less likely to confuse.

Also, your markdown formatting is hard to understand. You might want to use headers (bigger bolded text) to differentiate between sections. I didn't even understand you were using a numbered list for a while there.

If you want a numbered list, you can do that, but you probably want a table of contents for screen output and details. I've done that in a bunch of places (most pages, actually), in ostc-docs.